rust/core-lib/src/editor.rs

```diff
@@ -306,7 +306,7 @@ impl Editor {
     }
 
     pub(crate) fn delta_rev_head(&self, target_rev_id: RevToken) -> RopeDelta {
-        self.engine.delta_rev_head(target_rev_id)
+        self.engine.try_delta_rev_head(target_rev_id).unwrap()
```

Comment:
I _think_ it's probably best if we propagate this error up to the caller in this case. It's close, and I'm on the fence, but I think it's likely that future work will involve people using this function more, and I think that it should be communicated that there are cases that could fail.

Reply:
Makes sense. Also I noticed that there aren't any other `Result` types returned from `Editor` functions, is passing along an `Engine::Error` fine in this case or would we prefer to start supplying an `Editor::Error` to convert to?

Reply:
Good question. I agree it doesn't feel great to expose the engine error, and it also doesn't feel great to add a new error type for this single little case.

Another option would be to make this return an `Option<RopeDelta>`, and add a doc comment to this function explaining that it will return `None` if the revision is not found, and results should be checked if the revision is coming from a plugin?

Reply:
Cool so I made that an `Option` and added a doc, if there's some greater need for a new error type it will be easy to change in the future. Also added an error log where it's used in `event_context.rs`, but I'm unfamiliar with `SpecialEvent` handling so not sure if that's the right move

Reply:
Okay cool! I think we can actually skip the error handling in `SpecialEvent`, but the other change looks good; should be ready to merge after that little change (I'll do another quick pass now and double check, but I think this is gtg)